Feature/back office registration

timstone opened the pull request:
rip it a new one @Cruikshanks

timstone committed:
Add initial reg. of ind. and revocation via BO

This change adds the first back office features, which includes creating a registration of an exemption for an individual via the back office, and revoking an exemption.

Cruikshanks commented:
Working through it now @timstone . Think it'll be easier to do things in subsequent commits and we can then compare them to previous and discuss each in turn. So watch this space!

Cruikshanks committed:
Bring rubocop inline with FRA and resolve issues

This change brings the `.rubocop.yml` file in line with the [FRAE acceptance tests project](https://github.com/EnvironmentAgency/flood-risk-acceptance-tests). It also includes the fixes needed as a result of doing this.

Cruikshanks committed:
Rename step folders to not be case sensitive

It's considered better practise to avoid case sensitive folder and files names unless its required by your framework/tool, or its the convention for said file e.g. LICENSE and README's.

This change renames any folders which include uppercase characters to use use lowercase instead.

Cruikshanks committed:
Updating .gitignore

Cruikshanks committed:
Merge BO and FO general_steps.rb into one file

This change merges the BO and FO general steps into one file at the root of the steps folder. Currently the majority of the steps are general to both so this gives a starte for ten before then branching off into specific front and back office versions.

Cruikshanks committed:
Refactor BO page_objects to sep. login & search

Following the convention in FRAE, though the home page of the back office serves multiple functions and therefore we could group all logic into a single file, instead we separate them out into distinct files to make maintenance of them easier going forward.

This also involves updating the associated features and steps to match the new structure.
